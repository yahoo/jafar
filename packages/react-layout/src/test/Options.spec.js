/**
  * Copyright 2020, Verizon Media
  * Licensed under the terms of the MIT license. See LICENSE file in project root for terms.
  */

import React from 'react';
import { shallow } from 'enzyme';
import Options from '../components/Options/index';

describe('Options', () => {
  let props;

  beforeEach(() => {
    props = {
      options: [{
        label: 'Archive',
        onClick: jest.fn(),
      }],
    };
  });
  it('Should render - ok', () => {
    const component = shallow(getComponent(props));
    expect(component).toMatchSnapshot();
  });

  it('Should render - with action that has disabled truthy - ok', () => {
    props.options[0].disabled = true;
    const component = shallow(getComponent(props));
    expect(component).toMatchSnapshot();
  });

  function getComponent(props) {
    return (<Options {...props} />);
  }
});
