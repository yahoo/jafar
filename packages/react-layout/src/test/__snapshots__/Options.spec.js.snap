// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options Should render - ok 1`] = `
<div>
  <WithStyles(ForwardRef(IconButton))
    aria-haspopup="true"
    aria-label="Options"
    color="primary"
    onClick={[Function]}
  >
    <DehazeIcon />
  </WithStyles(ForwardRef(IconButton))>
  <Menu
    anchorEl={null}
    onClose={[Function]}
    open={false}
    options={
      Array [
        Object {
          "label": "Archive",
          "onClick": [MockFunction],
        },
      ]
    }
    style={
      Object {
        "maxHeight": 216,
        "width": 200,
      }
    }
  />
</div>
`;

exports[`Options Should render - with action that has disabled truthy - ok 1`] = `
<div>
  <WithStyles(ForwardRef(IconButton))
    aria-haspopup="true"
    aria-label="Options"
    color="primary"
    onClick={[Function]}
  >
    <DehazeIcon />
  </WithStyles(ForwardRef(IconButton))>
  <Menu
    anchorEl={null}
    onClose={[Function]}
    open={false}
    options={
      Array [
        Object {
          "disabled": true,
          "label": "Archive",
          "onClick": [MockFunction],
        },
      ]
    }
    style={
      Object {
        "maxHeight": 216,
        "width": 200,
      }
    }
  />
</div>
`;
